File: MLR-spring/src/main/java/com/mlr/controller/AdminController.java
package com.mlr.controller;

import com.mlr.service.AdminService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

@RestController
@RequestMapping("/api/admin")
@PreAuthorize("hasRole('ADMIN')")
public class AdminController {

    @Autowired
    private AdminService adminService;

    @GetMapping("/users")
    public ResponseEntity<?> getAllUsers(@RequestParam(required = false) String search) {

        return ResponseEntity.ok(adminService.getAllUsers(search));
    }

    @PostMapping("/users/{userId}/block")
    public ResponseEntity<?> blockUser(@PathVariable String userId) {
        boolean result = adminService.blockUser(userId);
        if (!result) return ResponseEntity.status(404).body("User not found");
        return ResponseEntity.ok(Map.of("message", "User blocked successfully"));
    }

    @PostMapping("/users/{userId}/unblock")
    public ResponseEntity<?> unblockUser(@PathVariable String userId) {
        boolean result = adminService.unblockUser(userId);
        if (!result) return ResponseEntity.status(440).body("User not found");
        return ResponseEntity.ok(Map.of("message", "User unblocked successfully"));
    }

    @PostMapping("/users/{userId}/reset-wallet")
    public ResponseEntity<?> resetWallet(@PathVariable String userId) {
        boolean result = adminService.resetWallet(userId);
        if (!result) return ResponseEntity.status(404).body("User wallet not found");
        return ResponseEntity.ok(Map.of("message", "Wallet reset successfully"));
    }

    @GetMapping("/analytics")
    public ResponseEntity<?> getSystemAnalytics() {
        return ResponseEntity.ok(adminService.getSystemAnalytics());
    }

    @GetMapping("/categories")
    @PreAuthorize("permitAll()") 
    public ResponseEntity<?> getCategories() {
        return ResponseEntity.ok(adminService.getCategories());
    }

    @GetMapping("/rules/status")
    public ResponseEntity<?> getRulesStatus() {
        return ResponseEntity.ok(adminService.getRulesStatus());
    }
}

File: MLR/AdminController.cs
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using MLR.Services;

namespace MLR.Controllers;

[ApiController]
[Route("api/[controller]")]
[Authorize(Roles = "ADMIN")]
public class AdminController : ControllerBase
{
    private readonly IAdminService _adminService;

    public AdminController(IAdminService adminService)
    {
        _adminService = adminService;
    }

    [HttpGet("users")]
    public async Task<IActionResult> GetAllUsers([FromQuery] string? search)
    {
        var users = await _adminService.GetAllUsersAsync(search);
        return Ok(users.Select(u => new
        {
            u.Id,
            u.Email,
            u.FirstName,
            u.LastName,
            IsBlocked = !u.IsActive,
            u.CreatedAt
        }));
    }

    [HttpPost("users/{userId}/block")]
    public async Task<IActionResult> BlockUser(string userId)
    {
        var result = await _adminService.BlockUserAsync(userId);
        if (!result)
            return NotFound("User not found");

        return Ok(new { Message = "User blocked successfully" });
    }

    [HttpPost("users/{userId}/unblock")]
    public async Task<IActionResult> UnblockUser(string userId)
    {
        var result = await _adminService.UnblockUserAsync(userId);
        if (!result)
            return NotFound("User not found");

        return Ok(new { Message = "User unblocked successfully" });
    }

    [HttpPost("users/{userId}/reset-wallet")]
    public async Task<IActionResult> ResetWallet(string userId)
    {
        var result = await _adminService.ResetWalletAsync(userId);
        if (!result)
            return NotFound("User wallet not found");

        return Ok(new { Message = "Wallet reset successfully" });
    }

    [HttpGet("analytics")]
    public async Task<IActionResult> GetSystemAnalytics()
    {
        var analytics = await _adminService.GetSystemAnalyticsAsync();
        return Ok(analytics);
    }

    [HttpGet("categories")]
    [AllowAnonymous]
    public async Task<IActionResult> GetCategories()
    {
        var categories = await _adminService.GetCategoriesAsync();
        return Ok(categories);
    }

    [HttpGet("rules/status")]
    public async Task<IActionResult> GetRulesStatus()
    {
        var rulesStatus = await _adminService.GetRulesStatusAsync();
        return Ok(rulesStatus);
    }
}